datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -174,8 +174,8 @@ pub fn can_coerce_from(type_into: &DataType, type_from: &DataType) -> bool {
                 | Float64
                 | Decimal128(_, _)
         ),
-        Timestamp(TimeUnit::Nanosecond, None) => {
-            matches!(type_from, Null | Timestamp(_, None))
+        Timestamp(TimeUnit::Nanosecond, _) => {
+            matches!(type_from, Null | Timestamp(_, _) | Date32)
```

Comment:
👍  (I have a draft PR in the works to support coercing from `utf8` as well https://github.com/apache/arrow-datafusion/pull/5117)